packages/azure-openapi-validator/autorest/src/spectral-plugin-func.ts

```diff
@@ -80,6 +83,30 @@ export async function spectralPluginFunc(initiator: IAutoRestPluginInitiator): P
   }
 }
 
+function ifStagingRunThenEnableRulesAsErrors(isStagingRun: boolean, initiator: IAutoRestPluginInitiator, ruleset: Ruleset) {
+  if (isStagingRun) {
+    // Here we ensure that in case of a staging run, all rules that are not enabled (i.e. not Warning or Error) are set to Error.
+    // The purpose of it is as follows: any such rule is assumed to be a rule that requires verification before
+    // it can be enabled in production. Hence we enable such rules in staging runs, to facilitate
+    // verifying them.
+    initiator.Message({
+      Channel: "information",
+      Text: "Detected Staging run. As a result, setting all non-Warning, non-Error spectral rules severity to Error.",
+    })
+    Object.values(ruleset.rules).forEach((rule) => {
+      // Note that DiagnosticsSeverity doesn't have a value for "off", and so we cannot simply check against it.
+      // The logic is more nuanced, where "off" ends up having value of -1 in DiagnosticSeverity.
+      // This is because function "getRuleSet" defined in this file calls into "set severity" in rule.ts which
+      // has the special handling logic. Source at:
+      // https://github.com/stoplightio/spectral/blob/44c40e2b7c9ea6222054da8700049b0307cc5f8b/packages/core/src/ruleset/rule.ts#L121
+      // where "off" is defined as -1 per:
+      // https://github.com/stoplightio/spectral/blob/44c40e2b7c9ea6222054da8700049b0307cc5f8b/packages/ruleset-migrator/src/transformers/rules.ts#L39
+      if ([DiagnosticSeverity.Warning, DiagnosticSeverity.Error].every((severity) => rule.severity != severity))
+        rule.severity = DiagnosticSeverity.Error
```

Comment:
Don't we have some rules that are never supposed to be Error? In such a case won't we have a lot of extra noise here because of this? 

Do we have any options that can mark a new rule as staging or some other indication that doesn't blindly promote everything to error?

Reply:
Re

> Don't we have some rules that are never supposed to be Error? In such a case won't we have a lot of extra noise here because of this?

From what I see [all rules are either warning or error](https://github.com/search?q=repo%3AAzure%2Fazure-openapi-validator+%22severity%3A+%22&type=code&p=1). There is one hit against severity of [recommended](https://github.com/Azure/azure-openapi-validator/blob/ce51affdc66115457bba7e7d5860ead92a7f92b4/packages/azure-openapi-validator/autorest/src/spectral-plugin-func.ts#LL178C59-L178C59) but this appears to be some ruleset context severity, which I think is most likely unused.

To get rid of the extra noise I could enable the rules as `Warning`s instead of `Error`s. Or  don't stress over it too much - it is staging after all, and we are planning to prevent it from ever becoming red.

Re

> Do we have any options that can mark a new rule as staging or some other indication that doesn't blindly promote everything to error?

I don't think you can do this, because of the spectral source code, as mentioned in this comment:

``` typescript
      // Note that DiagnosticsSeverity doesn't have a value for "off", and so we cannot simply check against it.
      // The logic is more nuanced, where "off" ends up having value of -1 in DiagnosticSeverity.
      // This is because function "getRuleSet" defined in this file calls into "set severity" in rule.ts which
      // has the special handling logic. Source at:
      // https://github.com/stoplightio/spectral/blob/44c40e2b7c9ea6222054da8700049b0307cc5f8b/packages/core/src/ruleset/rule.ts#L121
      // where "off" is defined as -1 per:
      // https://github.com/stoplightio/spectral/blob/44c40e2b7c9ea6222054da8700049b0307cc5f8b/packages/ruleset-migrator/src/transformers/rules.ts#L39
```

Reply:
I ended up applying this approach by stripping out `stagingOnly` just before the object is passed as an argument to `Ruleset` ctor. Later on, in the ctored `Ruleset`, the rules that were `stagingOnly` have their severity set to `off`.